pkg/3scale/amp/component/system.go

```diff
@@ -898,34 +912,65 @@ func (system *System) systemConfigVolumeMount() v1.VolumeMount {
 	}
 }
 
-func (system *System) appCommonContainerVolumeMounts(systemStorageReadonly bool) []v1.VolumeMount {
-	res := []v1.VolumeMount{}
-	if system.Options.pvcFileStorageOptions != nil {
-		res = append(res, system.systemStorageVolumeMount(systemStorageReadonly))
+func (system *System) fileStorageVolumeMountPath() string {
+	if system.fileStorageVolume().PersistentVolumeClaim != nil {
+		return system.pvcFileStorageVolumeMountPath()
+	}
+	if system.fileStorageVolume().Secret != nil {
+		return system.s3FileStorageVolumeMountPath()
 	}
+
+	// Should we panic?
+	panic("At least one of both file storages types has to be set")
+}
+
+func (system *System) s3FileStorageVolumeMountPath() string {
```

Comment:
At this moment we are mounting the S3 config in a DIFFERENT directory than the system extra-configs volume mount (system-config). By looking at documentation/links we might be able to share the same directory without overlapping/hidding other volumes for what I see here https://blog.sebastian-daschner.com/entries/multiple-kubernetes-volumes-directory via the subpath attribute.. I have not implemented it as a first approach. The main reason for not having done it as of now is because doing that would imply having to change/migrate the configuration of the system-config volume for already existing client installations.

Reply:
After talking with System, mounting the S3 config in a different directory than "system-extra-configs", where currently there are some configurations already, does not seem a good solution due to for every new secret/configmap with keys that we want to mount we would have to do it in a different directory, which would mean System code and Operator/Templates code would have to change every time.

I've explored the option to mount two ConfigMaps/Secrets in the same directory but this does not seem to be possible with how Kubernetes VolumeMounts work and the current Volume options we have. The reasons are:
* You cannot have multiple VolumeMounts in a container with the same "mountPath" attribute. The reason for that is that the mountPath attribute is used as a merge key when multiple volumeMounts have the same mountPath volume. This can be seen here: https://github.com/kubernetes/api/blob/7dc09db16fb8ff2eee16c65dc066c85ab3abb7ce/core/v1/types.go#L2181. So, if you try to create to volumeMounts with the same mountPath value, the two volumeMounts are merged into one.
* It seems there's sort of way to mount several fields of different Secrets/ConfigMaps in the same directory, but it implies changing the SubPath explicitely defining the key name of the desired Secret/ConfigMap to mount. This defeats precisely the purpose of making additions/changes without System/Operator/Templates being aware of it (and it also would imply a migration from templates/operator is needed). An example of this can be found here https://blog.sebastian-daschner.com/entries/multiple-kubernetes-volumes-directory. As you can observe, you have to explicitly write the secret attribute key name to mount in the subPath and in the mountPath attributes.
* We found the existence of "Projected" volumes that might seem interesting for the purpose but it is not enabled by default in Kubernetes 1.11 https://kubernetes.io/docs/concepts/storage/volumes/#projected so we cannot use it, as it's a version we must support (for OpenShift 3.11)